File: frameworks/dotnet/carter/Implementation/Routes/CompressedRoutes.cs
using Carter;

namespace Implementation.Routes;

/// <summary>
/// compressed: these routes answer like any other. ASP.NET Core's response compression,
/// installed on the whole application, gzips the answer when the request asks for it.
/// </summary>
public sealed class CompressedRoutes : ICarterModule
{
    public void AddRoutes(IEndpointRouteBuilder app)
    {
        app.MapGet("/compressed/small", (HttpResponse response, Payloads p) => Fresh(response, p.Small));

        app.MapGet("/compressed/large", (HttpResponse response, Payloads p) => Fresh(response, p.Large));
    }

    private static Payload Fresh(HttpResponse response, Payload payload)
    {
        Serial.Write(response);
        return payload;
    }
}

File: frameworks/java/micronaut/Implementation/src/main/java/implementation/routes/CompressedRoutes.java
package implementation.routes;

import implementation.Payload;
import implementation.Payloads;
import implementation.Serial;
import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;

/**
 * compressed: these routes answer like any other. The Netty server's compression, which
 * application.properties sets to gzip's fastest level for the whole server, gzips the answer when
 * the request asks for it.
 */
@Controller
public class CompressedRoutes {

    private final Payloads p;

    CompressedRoutes(Payloads p) {
        this.p = p;
    }

    @Get("/compressed/small")
    public HttpResponse<Payload> small() {
        return Serial.ok(p.small());
    }

    @Get("/compressed/large")
    public HttpResponse<Payload> large() {
        return Serial.ok(p.large());
    }
}
